gnd/src/main/java/com/google/android/gnd/ui/offlinearea/selector/OfflineAreaSelectorViewModel.java

```diff
@@ -16,27 +16,62 @@
 
 package com.google.android.gnd.ui.offlinearea.selector;
 
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
 import com.google.android.gms.maps.model.LatLngBounds;
 import com.google.android.gnd.repository.OfflineAreaRepository;
+import com.google.android.gnd.rx.Event;
+import com.google.android.gnd.rx.Schedulers;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
 import javax.inject.Inject;
 import timber.log.Timber;
 
 public class OfflineAreaSelectorViewModel extends AbstractViewModel {
 
-  private final OfflineAreaRepository offlineAreaRepository;
+  private final LiveData<Event<DownloadEvent>> downloadEvents;
+
+  public LiveData<Event<DownloadEvent>> getDownloadEvents() {
+    return this.downloadEvents;
+  }
+
+  enum DownloadEvent {
+    STARTED,
+    FAILURE
+  }
+
+  private final PublishSubject<DownloadEvent> downloadsPublishSubject = PublishSubject.create();
+  private final PublishSubject<LatLngBounds> downloadClickSubject = PublishSubject.create();
 
   @Inject
-  OfflineAreaSelectorViewModel(OfflineAreaRepository offlineAreaRepository) {
-    this.offlineAreaRepository = offlineAreaRepository;
+  OfflineAreaSelectorViewModel(OfflineAreaRepository offlineAreaRepository, Schedulers schedulers) {
+
+    disposeOnClear(
+        downloadClickSubject
+            .flatMap(
```

Comment:
I think maybe here you'd just want to use `flatMapCompletable` without the additional andThen?

Reply:
unfortunately, `flatMapCompletable` won't work here unless we call `onComplete` explicitly on the `PublishSubject`  from the docs

> Maps each element of the upstream Observable into CompletableSources, subscribes to them and waits until the upstream and all CompletableSources complete.

So, since a PublishSubject continually listens for clicks, it doesn't complete and the flatmap will never complete either (since it's perpetually waiting for the subject to complete). I added the comment just above the lambda to describe this predicament

Unless there's a subject we should use instead of `PublishSubject`? We could also force it to complete, but that would ensure the user can only ever click `Download` once, which is the current effect anyway. wdyt?

Reply:
I think if we're ok with forcing a single click to occur, the right approach is probably to use a `SingleSubject` for the clicks.